app/src/main/res/menu/main_menu_git.xml

```diff
@@ -29,6 +29,10 @@
         android:id="@+id/refresh"
         android:title="@string/refresh_list" />
 
+    <item
```

Comment:
Please also add this item to `main_menu_no_auth.xml`, which is our menu for configurations without write access to the GIt repository.

Reply:
The overflow menu is getting super crowded IMO so rather than add it here, it'd make much more sense to me if we get rid of the top hash from the Git utils screen and add a button to open the log from there.

Reply:
Agreed, that's even better. Viewing the Git log is likely not something one does often enough to warrant inclusion in this menu. The Git utils screen seems appropriate, we might even want to rename it into something like "Git log & utils".

Reply:
Thanks to both of you for your kind feedback! I'll gladly make the remaining changes.

I agree that the GitConfigActivity will probably be a more suitable place to access the log from. However, wouldn't you say that dropping the HEAD's commit hash there, which is accompanied either by the name of the current branch or the notice that HEAD is currently detached, would lose the user this information about the state of HEAD? Should we then keep it there or maybe provide this info another way?

If we keep it, how about changing the 'Commit hash 1234abcd (refs/heads/master [or DETACHED])' to be more direct Git wording - something like what you get on git-status: Either
- 'HEAD detached at 1234abcd' or
- 'On branch [refs/heads/]master'?

(Just an idea, though.)

![](https://user-images.githubusercontent.com/62889203/91472736-2f08cb80-e898-11ea-8099-f1fcd8c96e9a.png)

Reply:
I think you're headed in the right direction with your idea, so please go ahead and implement that. Also while at it, either drop the 'Hackish tools' label or just rename it to something more sensible. I think these have been improved and rewritten enough to not be 'hacks' anymore.